providers/defltprov.c

```diff
@@ -396,6 +396,26 @@ static const OSSL_ALGORITHM deflt_signature[] = {
     { PROV_NAMES_DSA, "provider=default", ossl_dsa_signature_functions },
 #endif
     { PROV_NAMES_RSA, "provider=default", ossl_rsa_signature_functions },
+#if !defined(OPENSSL_NO_MD5) && !defined(FIPS_MODULE)
+    { PROV_NAMES_RSA_MD5, "provider=default", ossl_rsa_md5_signature_functions },
+#endif
```

Comment:
I'd suggest skipping RSA-MD5 for now. It is a pretty much legacy algorithm and should be probably implemented in the legacy provider only. Once we add it here, it will be hard to remove it.